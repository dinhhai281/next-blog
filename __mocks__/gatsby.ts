import React from 'react';
const gastby = jest.requireActual('gatsby');

module.exports = {
  ...gastby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({ activeClassName, activeStyle, getProps, innerRef, partiallyActive, ref, replace, to, ...rest }) =>
        React.createElement('a', { ...rest, href: to })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
