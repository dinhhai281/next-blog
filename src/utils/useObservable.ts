import { useEffect, SetStateAction, Dispatch } from 'react';
import { Observable } from 'rxjs';

export const useObservable = <T extends unknown>(input$: Observable<T>, setter: Dispatch<SetStateAction<T>>) => {
  useEffect(() => {
    const subscription = input$.subscribe(setter);

    return () => subscription.unsubscribe();
  }, [input$]);
};
