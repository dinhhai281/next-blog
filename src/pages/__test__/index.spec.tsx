import { renderWithProviders, setupIntersectionObserverMock, siteMetadataStub } from '@app/utils/testing';
import { useStaticQuery } from 'gatsby';
import Home from 'pages';
import React from 'react';

describe('Home', () => {
  beforeAll(() => {
    setupIntersectionObserverMock();
  });

  it('render correctly', () => {
    (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
      site: { siteMetadata: siteMetadataStub },
    }));

    const { asFragment } = renderWithProviders(
      <Home
        data={{
          allMarkdownRemark: {
            nodes: [],
          },
          allGithubData: {
            nodes: [
              {
                data: {
                  viewer: {
                    avatarUrl: 'avatarUrl',
                    email: 'email',
                    login: 'login',
                    name: 'name',
                  },
                },
              },
            ],
          },
        }}
      ></Home>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
