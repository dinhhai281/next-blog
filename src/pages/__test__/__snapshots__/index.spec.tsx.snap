// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-3n56dt"
  >
    <main
      class="css-1nwsnwk"
    >
      <div
        class="css-lzuhz1"
      >
        <p
          class="chakra-text css-1hwj3aj"
        >
          home.greeting
        </p>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-zvqp6p"
      >
        <p
          class="chakra-text css-1efqfyl"
        >
          home.intro
        </p>
      </div>
      <div
        class="css-3krwau"
      >
        <span
          class="chakra-avatar css-wbs104"
        >
          <div
            aria-label="Hai Nguyen"
            class="chakra-avatar__initials css-0"
          >
            HN
          </div>
        </span>
        <div
          class="css-17xejub"
        />
        <button
          class="chakra-button css-q4uigm"
          type="button"
        >
          <span
            class="chakra-button__icon css-19u2hmv"
          >
            <img
              alt="github logo"
              aria-hidden="true"
              class="chakra-image__placeholder css-1vx4smm"
              focusable="false"
            />
          </span>
          Github
        </button>
      </div>
      <div
        class="chakra-stack css-tzcfgn"
      >
        <div
          class="css-0"
        >
          <div
            class="css-175wqfc"
          >
            <p
              class="chakra-text css-1pc31l6"
            >
              Programming
            </p>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
